pallets/msa/src/lib.rs

```diff
@@ -297,6 +297,32 @@ pub mod pallet {
 	pub type ApprovedLogos<T: Config> =
 		StorageMap<_, Twox64Concat, LogoCid<T>, BoundedVec<u8, T::MaxLogoSize>, OptionQuery>;
 
+	/// Monotonically increasing index of applications for a given provider.
+	/// Starts at 0 when provider first gets an application approved.
+	#[pallet::storage]
+	#[pallet::getter(fn next_application_index)]
+	pub(super) type NextApplicationIndex<T: Config> =
+		StorageMap<_, Twox64Concat, ProviderId, ApplicationIndex, ValueQuery>;
```

Comment:
this will ensure O(1) for next application index for a provider id

Reply:
Yeah--was thinking about other ways to handle this. I think the issue is not so much that we need O(1) (since # of applications per provider will usually be very low, like single digits), but that other solutions (say, storing the Applications in a BoundedVec tied to the ProviderId) have other drawbacks, like size of read and reusing indices.

This is probably the best approach, overall.